### PR TITLE
Release: 2.6.0

### PR DESCRIPTION
**Pull request by nerrad**
This is the release pull request for `v2.6.0` of the WooCommerce Blocks plugin.

> NOTE: This release should _not_ be added to WooCommerce core because [we don't have gating implemented for Cart & Checkout blocks yet](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2451) (should be implemented in 2.7.0).

## Communication

* [x] What is being introduced in this release?

- This release introduces **Cart and Checkout blocks** for increased visibility and wider testing as a part of the consideration of eventually introducing these blocks to WooCommerce core.  
- There are a number of improvements to existing blocks as noted in the changelog.
- The Store REST API has been refactored with new routes added. As well all [older endpoints were migrated to use the new routes on the Store routes](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2282). None of these REST API endpoints are considered public and thus there is no related versioning. Notably, we also introduced a new `X-WC-Store-API-Nonce` header (and nonce!) to our store endpoints that adds CSRF protection. This is particularly important for Cart and Checkout related write requests for a session.
- Introduction of a new status for orders ("draft"). (Introduced in [this pull](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1425))

**Note:** For folks that have been testing the beta and release versions of the cart and checkout blocks, it's possible that they could experience [this reported issue](https://github.com/woocommerce/woocommerce-admin/issues/4447) for the new orders list table in the WC-Admin. If this happens, the only way to fix currently is to manually delete orders in the `wc_order_stats` table that don't have corresponding orders in the `wp_posts` table by executing this query (note if this is a db you care about, as a safety precaution you will want to backup first):

```sql
DELETE stats FROM wp_wc_order_stats stats LEFT JOIN wp_posts posts ON stats.order_id = posts.ID WHERE posts.ID IS NULL;
```

The above query assumes `wp_` is the prefix for tables in the db - change that to whatever the prefix is if it's different. You can also replace `DELETE stats` with `SELECT stats.*` if you want to see what would be affected first.

- Introduction of a [new Reserved Stock class and database table](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1778) to prevent race conditions during checkout. [Equivalent WooCommerce core change here](https://github.com/woocommerce/woocommerce/pull/25708). Note also the [table creation validation for install routine added](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2287). **Note:** this just adds the table and if it fails being created, the merchant's store should still function as before (but without the race condition protection for low stock on busy stores). This also means that reverting back to an earlier version of the plugin should work okay (the only side-effect being that any instances of the Cart and Checkout block will display no content).

* [x] If this release has potentially **high customer impact**, is a blog post (section) ready? Provide a link if applicable.

**Impacts for this release on merchants**

- Cart and Checkout blocks - [blog post is ready](https://woocommerce.wordpress.com/?p=6384). _Note:_ with the plugin release we are positioning this as a "preview" of the Cart and Checkout blocks for folks to experiment with. They are still being evaluated for potential inclusion to core.
- Release post for develop.woocommerce.com [is written](https://woocommerce.wordpress.com/?p=6577).
- All Products (and related filter blocks): A number of changes were made to the underlying logic and code being used. The impact is on ensuring existing implementations of the blocks still work after upgrading.
- HTML editing is [no longer supported in several blocks](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1395). This means that any merchants that have blocks with edited HTML will "still work" but they will no longer be able to re-edit.
- Product categories block [now supports category images](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1739).
- Wrote a post for [Woo Happiness team as well](https://woosupport.wordpress.com/?p=27672).

**Blocks with a changeset**

Besides the above notables, here's a list of other blocks that are impacted by changes in this release.

- Product Search
- Featured Product
- All "Product Grid type blocks", that includes Handpicked Products, Product Best Sellers, Product Category, Product New, Product On Sale, Products by Attribute, Product Tag, Product Top Rated. **Note:** Product Grid blocks now default to a default of 3 rows (instead of one) for the initial setting. This means for SSR blocks, existing blocks rows will change to 3 unless the user has already changed them away from 1. More details in [this pull](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1613).

* [x] Is the **relevant developer documentation updated**? Please provide links if applicable.

- [Documentation for Store REST API](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/master/src/StoreApi). **Note** this is still internal use only. It is not a public API for consumption.
- Various [block client apis](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/master/docs/block-client-apis).
- Current [extensibility docs](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/master/docs/extensibility)  - Note: currently this applies only to Cart and Checkout blocks and payment methods. However we are planning on expanding these docs over time.

## Quality

* [x] Have you **written tests** to cover the changes?
     * [x] Unit tests
     * [x] E2E tests
     * [x] for each supported WordPress and WooCommerce core versions.

**Note**: Our automated test coverage is still being improved. 

- The REST API Store routes have full coverage on the routes.
- There is very little automated test coverage of Cart and Checkout blocks.
- There is limited coverage of other blocks.

What this means practically is that a lot of heavy lifting is being done by manual testing for now.

* [x] Have you tested on the applicable platforms?
     * [x] mobile
     * [x] desktop
* [x] Does this release affect API's and conform to API versioning policy?

 Current REST API is internal only. The public API surface for integrations etc has not changed. The public API surface for payment method integrations is **still subject to change**.

* [x] Have you considered the impact of the changes in the release on **other extensions** and **backward compatibility**?
     * [x] Does the release change the signature of any public methods or functions?
     * [x] Are filters or hooks affected by the PR code changes?
* [x] Does the release include a changelog entry in the readme.txt?
* [x] Please provide a link to **testing instructions** for this release. **See [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2554#issuecomment-633681094https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2554#issuecomment-633681094)**

* [x] Have you reviewed impacts to performance (bundle sizes, query time etc)?

**Note:** As noted by @senadir [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2554#issuecomment-633665458), there is some impact to bundle sizes for frontend js built for the All Products and Filter blocks. We are [aware of this impact](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2280) which is caused by some dependencies pulling in DashIcons and have a potential fix that is not ready for this release. The target is `2.7.0` for having this fixed so that when the feature plugin is updated in WooCommerce core this impact is not included there.

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -120,6 +120,41 @@ Release and roadmap notes available on the [WooCommerce Developers Blog](https:/
...
+- dev: All Products Block: Update sorting labels to match frontend options [#2462](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2462)
+- dev: Change PropType validation for Icon component [#1737](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1737)
+- dev: Changed default rows and columns for product grid blocks to 3x3. [#1613](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1613)
+- dev: Check for instance of WP_Block in render_callback [#2258](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2258)
+- dev: Devs: `ENABLE_REVIEW_RATING` setting was renamed to `REVIEW_RATINGS_ENABLED` and now it also verifies reviews are enabled, to better match WooCommerce API. [#1374](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1374)
+- dev: Fix price filtering when stored prices do not match displayed prices (determined by tax settings). This fix will apply once WooCommerce 3.10 is released. [#1612](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1612)
```
Comment (Aljullu) [inline readme.txt:138]:
```suggestion
- dev: Fix price filtering when stored prices do not match displayed prices (determined by tax settings). [#1612](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1612)
```

File: readme.txt [hunk 2]
```diff
@@ -120,6 +120,41 @@ Release and roadmap notes available on the [WooCommerce Developers Blog](https:/
...
+- dev: HTML editing is no longer supported in several blocks. [#1395](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1395)
+- dev: Implement __experimentalCreateInterpolateElement for translations. [#1736](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1736)
+- dev: Load WooCommerce Core translations for 'Sale!'  and some other strings if translations are unavailable for WooCommerce Blocks. [#1694](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1694)
```
Comment (Aljullu) [inline readme.txt:141]:
```suggestion
- dev: Load WooCommerce Core translations for 'Sale!' and some other strings if translations are unavailable for WooCommerce Blocks. [#1694](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1694)
```

**Comment by senadir** [comment]
This is the result of the bundle change of files from the latest 2.5 release to master, this only includes *-frontend.js files

| Bundle                             | `release/2.5` | `master`  | Diff    |
| ---------------------------------- | ----------- | ------- | ------- |
| `active-filters-frontend.js`   | 5.27KB      | 7.09KB  | 1.81KB  |
| `all-products-frontend.js`    | 14.37KB     | 70.12KB | 55.75KB |
| `attribute-filter-frontend.js` | 6.68KB      | 16.31KB | 9.63KB  |
| `price-filter-frontend.js`     | 7.1KB       | 13.53KB | 6.43KB  |
| `reviews-frontend.js`          | 7.53KB      | 8.63KB  | 1.1KB   |

This is a full list of all files that existed or didn't exist.

<details><summary><strong>All Files</strong></summary>


| Bundle                                  | Release 2 5 | Master   | Diff     |
| --------------------------------------- | ----------- | -------- | -------- |
| `active-filters-frontend.js`            | 5.27KB      | 7.09KB   | 1.81KB   |
| `active-filters.js`                     | 5.93KB      | 7.73KB   | 1.8KB    |
| `all-products-frontend.js`              | 14.37KB     | 70.12KB  | 55.75KB  |
| `all-products.js`                       | 63.07KB     | 18.73KB  | -44.34KB |
| `all-reviews-legacy.js`                 | 9.84KB      | 10.13KB  | 293B     |
| `all-reviews.js`                        | 9.93KB      | 10.41KB  | 488B     |
| `attribute-filter-frontend.js`          | 6.68KB      | 16.31KB  | 9.63KB   |
| `attribute-filter.js`                   | 8.19KB      | 11.21KB  | 3.02KB   |
| `block-error-boundary-legacy.js`        | 811B        | 813B     | 2B       |
| `block-error-boundary.js`               | 809B        | 808B     | -1B      |
| `blocks-legacy.js`                      | 2.18KB      | 2.88KB   | 719B     |
| `blocks.js`                             | 2.18KB      | 2.88KB   | 724B     |
| `cart-frontend.js`                      | 0B          | 110.22KB | 110.22KB |
| `cart.js`                               | 0B          | 31.42KB  | 31.42KB  |
| `checkout-frontend.js`                  | 0B          | 126.21KB | 126.21KB |
| `checkout.js`                           | 0B          | 36.98KB  | 36.98KB  |
| `custom-select-control-style-legacy.js` | 0B          | 818B     | 818B     |
| `custom-select-control-style.js`        | 0B          | 813B     | 813B     |
| `featured-category-legacy.js`           | 141.48KB    | 142.41KB | 954B     |
| `featured-category.js`                  | 141.65KB    | 142.38KB | 751B     |
| `featured-product-legacy.js`            | 56.38KB     | 9.3KB    | -47.08KB |
| `featured-product.js`                   | 56.81KB     | 9.6KB    | -47.21KB |
| `handpicked-products-legacy.js`         | 5.78KB      | 6.75KB   | 1000B    |
| `handpicked-products.js`                | 5.79KB      | 7.02KB   | 1.22KB   |
| `panel-style-legacy.js`                 | 0B          | 806B     | 806B     |
| `panel-style.js`                        | 0B          | 801B     | 801B     |
| `price-filter-frontend.js`              | 7.1KB       | 13.53KB  | 6.43KB   |
| `price-filter.js`                       | 8.13KB      | 9.76KB   | 1.63KB   |
| `product-best-sellers-legacy.js`        | 5.76KB      | 6.83KB   | 1.07KB   |
| `product-best-sellers.js`               | 5.78KB      | 7.09KB   | 1.31KB   |
| `product-categories-legacy.js`          | 2.88KB      | 3.17KB   | 298B     |
| `product-categories.js`                 | 2.88KB      | 3.16KB   | 284B     |
| `product-category-legacy.js`            | 6.68KB      | 7.72KB   | 1.04KB   |
| `product-category.js`                   | 6.7KB       | 7.98KB   | 1.28KB   |
| `product-list-style-legacy.js`          | 813B        | 814B     | 1B       |
| `product-new-legacy.js`                 | 5.95KB      | 6.98KB   | 1.04KB   |
| `product-new.js`                        | 5.96KB      | 7.25KB   | 1.29KB   |
| `product-on-sale-legacy.js`             | 6.37KB      | 7.34KB   | 996B     |
| `product-on-sale.js`                    | 6.4KB       | 7.64KB   | 1.24KB   |
| `product-search-legacy.js`              | 2.83KB      | 3.08KB   | 259B     |
| `product-search.js`                     | 2.85KB      | 3.31KB   | 474B     |
| `product-tag-legacy.js`                 | 5.61KB      | 5.96KB   | 362B     |
| `product-tag.js`                        | 5.63KB      | 6.21KB   | 592B     |
| `product-top-rated-legacy.js`           | 5.76KB      | 6.95KB   | 1.2KB    |
| `product-top-rated.js`                  | 5.78KB      | 7.22KB   | 1.44KB   |
| `products-by-attribute-legacy.js`       | 6.6KB       | 7.7KB    | 1.1KB    |
| `products-by-attribute.js`              | 6.62KB      | 7.95KB   | 1.33KB   |
| `reviews-by-category-legacy.js`         | 11.77KB     | 12.07KB  | 307B     |
| `reviews-by-category.js`                | 11.89KB     | 12.42KB  | 543B     |
| `reviews-by-product-legacy.js`          | 12.93KB     | 13.53KB  | 619B     |
| `reviews-by-product.js`                 | 13.09KB     | 13.86KB  | 783B     |
| `reviews-frontend-legacy.js`            | 7.21KB      | 7.84KB   | 646B     |
| `reviews-frontend.js`                   | 7.53KB      | 8.63KB   | 1.1KB    |
| `snackbar-notice-style-legacy.js`       | 0B          | 817B     | 817B     |
| `snackbar-notice-style.js`              | 0B          | 812B     | 812B     |
| `spinner-style-legacy.js`               | 0B          | 811B     | 811B     |
| `spinner-style.js`                      | 0B          | 804B     | 804B     |
| `vendors-legacy.js`                     | 0B          | 366.89KB | 366.89KB |
| `vendors-style-legacy.js`               | 0B          | 153B     | 153B     |
| `vendors-style.js`                      | 0B          | 146B     | 146B     |
| `vendors.js`                            | 0B          | 462.44KB | 462.44KB |
| `wc-blocks-data.js`                     | 3.87KB      | 6.95KB   | 3.08KB   |
| `wc-blocks-middleware.js`               | 0B          | 968B     | 968B     |
| `wc-blocks-registry.js`                 | 879B        | 1.79KB   | 949B     |
| `wc-payment-method-cheque.js`           | 0B          | 834B     | 834B     |
| `wc-payment-method-paypal.js`           | 0B          | 867B     | 867B     |
| `wc-payment-method-stripe.js`           | 0B          | 11.66KB  | 11.66KB  |
| `wc-settings.js`                        | 2.13KB      | 2.12KB   | -14B     |

</details>

**Comment by nerrad** [comment]
Follow-up tasks for after the release is tagged and published to wp.org

* [ ] Publish Cart and Checkout blocks post (and then add link to this pull description).
* [ ] Publish 2.6.0 release post including various DEV notes (and then add link to this pull description).
* [ ] Publish a post for Woo Happiness engineers highlighting changes in this release.

**Comment by nerrad** [comment]
## Testing Notes and ZIP for testing

_in progress_

**Comment by nerrad** [comment]
> I noticed one issue, though this might not be a problem. With the Filter by Attribute block, the new Dropdown mode is available for both Query Types (Or and And). It works correctly, so this seems fine to me. (Testing instructions said it should only be available for Or.)

@Aljullu can you comment on this. Was there a follow-up pull that enabled support for "And" that got missed in the changelog or is it something already available in the 2.5 release branch?

**Comment by Aljullu** [comment]
@nerrad correct, the dropdown for `AND` query type was introduced in #1309. It has the `skip-changelog` label, probably that's why it was missed.

So I think we need to update the testing instructions so both modes are tested.

**Comment by nerrad** [comment]
2.6.0 is released and follow-ups around documenting changes in process are [handled in this pull](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2598).

**Pull request closed without merging**